### PR TITLE
1702 - ids-swaplist search filter

### LINKED ISSUE #1702: IdsSwapList: Add Filtering feature
**Is your feature request related to a problem or use case? Please describe.**
We are converting the UIs for our integration components in M3 from SoHo to Web Components. The UIs for the components Data Lake Publisher and Event Hub use Swap Lists with many selectable items (I currently have 3657 in DLP and 744 in EH, customers will have more). It is crucial to be able to filter the left and right lists, which you can do in SoHo, see https://latest-enterprise.demo.design.infor.com/components/swaplist/example-search.html. Currently this functionality is missing for Web Components. We can implement a workaround, but we want to release the UIs using standard Web Component functionality.

**Describe the solution you'd like**
We need Swap Lists to provide filtering/search for each list, as it works for SoHo. The filtering should be case insensitive and only show items in the list containing the search string. The search string should be shown in bold style for every matching item. After the item is swapped to the other list, the filtering field should automatically be cleared, and all items for the list should be shown again.
Default text ("Search Available", "Search Selected") should be available.
There should be an "X" button to remove the filtering and clear the search field if a search string is given.

**Describe alternatives you've considered**
We can implement a workaround (I don't know how it works technically, I'm not a developer, contact Kenneth Acejo), but we want to release the UIs using standard Web Component functionality. From an UX point of view it has to work exactly as it works for SoHo. The UI layout can be somewhat adopted to "Web Component style", if necessary.

**Additional context**
I work with Infor M3 integration components.
Screenshot of one of the current UIs:
image

### PR DESCRIPTION
**Pull request by clayinfor**
**Explain the details for making this change. What existing problem does the pull request solve?**

Added search feature to `IdsSwapList`.

**Related github/jira issue (required)**:

Closes #1702 

**Steps necessary to review your pull request (required)**:

1. Check out this branch and run: `nvm use && npm install && npm run start`
2. Go to: http://localhost:4300/ids-swaplist/search.html
3. Ensure search works for each column even after moving things around.
4. Ensure search fields have placeholders that match the titles of their column
5. Ensure case-sensitive swaplist works properly. 

**Included in this Pull Request**:
- [ ] Some documentation for the feature.
- [ ] A test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: src/components/ids-list-view/ids-list-view-search-mixin.ts [hunk 1]
```diff
@@ -308,7 +308,7 @@ const IdsListViewSearchMixin = <T extends Constraints>(superclass: T) => class e
    */
   #handleSearch(term: string): void {
     const lv = this as unknown as IdsListView;
-    if (!lv.datasource) return;
+    // if (!lv.datasource) return;
```
Comment (tmcconechy) [inline src/components/ids-list-view/ids-list-view-search-mixin.ts:311]: Could just remove

File: src/components/ids-list-view/ids-list-view-search-mixin.ts [hunk 2]
```diff
@@ -384,9 +384,10 @@ const IdsListViewSearchMixin = <T extends Constraints>(superclass: T) => class e
   /**
    * Set search term highlight
    * @param {string | HTMLElement | object} item The from data item
+   * @param {string | null} haystack item text to search
```
Comment (tmcconechy) [inline src/components/ids-list-view/ids-list-view-search-mixin.ts:387]: What is a haystack? Maybe need a better term here?

**Review by tmcconechy — COMMENTED** [review]
1. looks like some lint issues
2. a couple minor things inline
3. Can we style it like https://main-enterprise.demo.design.infor.com/components/swaplist/example-search.html (not super important but we could do it be overriding the particular tokens for search field in the host template)

**Comment by clayinfor** [comment]
hey @tmcconechy , I've resolve the issues you mentioned except for styling... think will be best to revisit styling in another ticket.

**Review by tmcconechy — APPROVED** [review]

**Comment by tmcconechy** [comment]
@clayinfor added styles

**Review by jdilag90 — APPROVED** [review]
Nice!

**Pull request merged**